app/models/task.rb

```diff
@@ -1,6 +1,6 @@
 class Task < ApplicationRecord
   belongs_to :list
-  has_many :labels
+  belongs_to :labels, optional: true
```

Comment:
IMHO each task `has_one :label`

Reply:
According to my understanding, the `has_one` association specifies a one-to-one association with another class, but in our case we need a one-to-many association with the label class.
However I made a mistake writing `labels` (in plural) instead of `label`, so I will fix it.